src/date-range-picker/calendar/grids/grid.tsx

```diff
@@ -0,0 +1,335 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useMemo } from 'react';
+import clsx from 'clsx';
+import {
+  addDays,
+  addMonths,
+  addQuarters,
+  addWeeks,
+  getDaysInMonth,
+  isAfter,
+  isBefore,
+  isLastDayOfMonth,
+  isSameDay,
+  isSameMonth,
+  isSameYear,
+  isThisMonth,
+  isToday,
+} from 'date-fns';
+import { getCalendarMonth } from 'mnth';
+
+import useCalendarGridRows from '../../../calendar/grid/use-calendar-grid-rows';
+import { useInternalI18n } from '../../../i18n/context';
+import ScreenreaderOnly from '../../../internal/components/screenreader-only';
+import { formatDate } from '../../../internal/utils/date-time';
+import { normalizeLocale } from '../../../internal/utils/locale';
+import { DateRangePickerProps } from '../../interfaces';
+import { GridCell } from './grid-cell';
+import { GridProps } from './interfaces';
+import { renderDateAnnouncement, renderDayName } from './intl';
+
+import testutilStyles from '../../test-classes/styles.css.js';
+import styles from './styles.css.js';
+
+/**
+ * Calendar grid supports two mechanisms of keyboard navigation:
+ * - Native screen-reader table navigation (semantic table markup);
+ * - Keyboard arrow-keys navigation (a custom key-down handler).
+ *
+ * The implementation largely follows the w3 example (https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/datepicker-dialog) and shares the following issues:
+ * - (table navigation) Chrome+VO - weekday is announced twice when navigating to the calendar's header;
+ * - (table navigation) Safari+VO - "dimmed" state is announced twice;
+ * - (table navigation) Firefox/Chrome+NVDA - cannot use table navigation if any cell has a focus;
+ * - (keyboard navigation) Firefox+NVDA - every day is announced as "not selected";
+ * - (keyboard navigation) Safari/Chrome+VO - weekdays are not announced;
+ * - (keyboard navigation) Safari/Chrome+VO - days are not announced as interactive (clickable or selectable);
+ * - (keyboard navigation) Safari/Chrome+VO - day announcements are not interruptive and can be missed if navigating fast.
+ */
+
+export function Grid({
+  baseDate,
+  selectedStartDate,
+  selectedEndDate,
+  rangeStartDate,
+  rangeEndDate,
+  focusedDate,
+
+  focusedDateRef,
+
+  onSelectDate,
+  onGridKeyDownHandler,
+  onFocusedDateChange,
+
+  isDateEnabled,
+  dateDisabledReason,
+
+  locale,
+  todayAriaLabel,
+  currentMonthAriaLabel,
+  ariaLabelledby,
+  className,
+  startOfWeek = 0,
+  granularity = 'day',
+}: GridProps) {
+  const baseDateTime = baseDate?.getTime();
+  // `baseDateTime` is used as a more stable replacement for baseDate
+  const weeks = useMemo<Date[][]>(
+    () => getCalendarMonth(baseDate, { firstDayOfWeek: startOfWeek }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Instead of this ignore, I would simply convert `baseDateTime` back to date

```suggestion
    () => getCalendarMonth(new Date(baseDateTime), { firstDayOfWeek: startOfWeek }),
```

If you are inside memo and you assume the value does not change often, this double Date -> number -> Date conversion costs nothing

However, not ignoring linter will avoid future bugs, if anything changes here

Reply:
Agreed